addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestBridgeHandler.java

```diff
@@ -8,127 +8,131 @@
  */
 public class NestBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(NestBridgeHandler.class);
 
-    private Logger logger = LoggerFactory.getLogger(NestBridgeHandler.class);
-
-    private List<NestDeviceAddedListener> listeners = new ArrayList<NestDeviceAddedListener>();
-
-    // Will refresh the data each time it runs.
-    private Runnable pollingRunnable = new Runnable() {
-        @Override
-        public void run() {
-            refreshData();
-        }
-    };
-
-    private ScheduledFuture<?> pollingJob;
-    private NestAccessToken accessToken;
-    private List<NestUpdateRequest> nestUpdateRequests = new ArrayList<>();
-    private TopLevelData lastDataQuery;
-    private HttpClient httpClient;
+    private final List<NestDeviceDataListener> listeners = new ArrayList<>();
+    private final List<NestUpdateRequest> nestUpdateRequests = new CopyOnWriteArrayList<>();
+    private final Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").create();
 
-    private final GsonBuilder builder;
+    private ScheduledFuture<?> refreshDataJob;
+    private NestAuthorizer authorizer;
+    private NestBridgeConfiguration config;
+    private ScheduledFuture<?> sender;
+    private String redirectUrl;
 
     /**
-     * Creates the bridge handler to connect to nest.
+     * Creates the bridge handler to connect to Nest.
      *
-     * @param bridge The bridge to connect to nest with.
+     * @param bridge The bridge to connect to Nest with.
      */
     public NestBridgeHandler(Bridge bridge) {
         super(bridge);
-        builder = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
     }
 
     /**
-     * Initialize the connection to nest.
+     * Initialize the connection to Nest.
      */
     @Override
     public void initialize() {
-        logger.debug("Initialize the Nest bridge handler");
+        logger.debug("Initializing Nest bridge handler");
+
+        config = getConfigAs(NestBridgeConfiguration.class);
+        authorizer = new NestAuthorizer(config);
+
+        logger.debug("Product ID      {}", config.productId);
+        logger.debug("Product Secret  {}", config.productSecret);
+        logger.debug("Pincode         {}", config.pincode);
 
-        // Create the jetty client and configure for https.
-        SslContextFactory sslContextFactory = new SslContextFactory();
-        httpClient = new HttpClient(sslContextFactory);
-        httpClient.setConnectTimeout(30000);
         try {
-            httpClient.start();
-        } catch (Exception e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                    "Error starting HTTP client " + e.getMessage());
-            return;
+            logger.debug("Access Token    {}", getExistingOrNewAccessToken());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Starting poll query");
```

Comment:
OFFLINE means we cannot access the server - here we are just trying to do so. So either don't set a status at all or use UNKNOWN here.

Reply:
UNKNOWN now.